config/7.2.0/securebanking/ig/scripts/groovy/SignEventsResponse.groovy

```diff
@@ -0,0 +1,102 @@
+    }
+}
+
+Error response example:
+{
+    "Code": "500",
+    "Id": "ede36b552dc951d9836a127f16a7c033",
+    "Message": "[Status: 500 Internal Server Error]",
+    "Errors": [
+        {
+            "ErrorCode": "UK.OBIE.UnexpectedError",
+            "Message": "Internal error [Unknown Signing Algorithm]"
+        }
+    ]
+}
+
+Validate signature
+- You can validate the JWT SET signature in: https://jwt.davetonge.co.uk/
+- Copy a JWT 'set' value and paste in the web page.
+- JWKs URL endpoint to validate the signature: https://keystore.openbankingtest.org.uk/0015800001041REAAY/0015800001041REAAY.jwks
+ */
+
+def fapiInteractionId = request.getHeaders().getFirst("x-fapi-interaction-id");
+if (fapiInteractionId == null) fapiInteractionId = "No x-fapi-interaction-id"
+SCRIPT_NAME = "[SignEventsResponse] (" + fapiInteractionId + ") - "
+
+Map<String, Object> critClaims = Map.of(
+        "http://openbanking.org.uk/iat", System.currentTimeMillis() / 1000,
+        "http://openbanking.org.uk/iss", aspspOrgId,
+        "http://openbanking.org.uk/tan", "openbanking.org.uk")
+
+next.handle(context, request).thenOnResult({ response ->
```

Comment:
thenOnResult needs to be changed to thenAsync, as within the handler we need to make a further async calls and return a Promise.